File: Algorithm/baekjoon/bj_1043.py
# N 사람 수, M 파티 수 
N,M=map(int,input().split())

# truth : 진실을 아는 사람
truth = set(input().split()[1:])

# members : 파티의 사람 
members = [set(input().split()[1:]) for _ in range(M)]

# 모든 파티의 사람들을 다 합친 다음,
for _ in range(M):
    for member in members:
        if member&truth: truth |= member

# 진실을 아는, 알게되는 애들이랑만 얘기 안하면 된다.
cnt = 0
for member in members:
    if not member&truth : cnt += 1
print(cnt)

File: Algorithm/baekjoon/java/src/bj_1043.java
import java.io.*;
import java.util.*;

public class bj_1043 {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;
        // N,M
        st = new StringTokenizer(br.readLine());
        int N = Integer.parseInt(st.nextToken());
        int M = Integer.parseInt(st.nextToken());

        // T, truth
        Set<Integer> truth = new HashSet<>();
        st = new StringTokenizer(br.readLine());
        int T = Integer.parseInt(st.nextToken());

        for (int i=0; i<T; i++) {
            truth.add(Integer.parseInt(st.nextToken()));
        }

        // member, members
        Set<Integer>[] members = new Set[M];
        for (int i=0; i<M; i++) {
            st = new StringTokenizer(br.readLine());
            int mem = Integer.parseInt(st.nextToken());
            Set<Integer> member = new HashSet<>();
            for (int j=0; j<mem; j++) {
                member.add(Integer.parseInt(st.nextToken()));
            }
            members[i] = member;
        }
        // 합친 세트 만들기
        for (int i=0; i<M; i++) {
            for (Set<Integer> member : members) {
                Set<Integer> tmp = new HashSet<>(member);
                tmp.retainAll(truth);
                if (!tmp.isEmpty()) {
                    truth.addAll(member);
                }
            }
        }
        int cnt = 0;
        for (Set<Integer> member : members) {
            member.retainAll(truth);
            if (member.isEmpty()) cnt++;
        }
        System.out.println(cnt);
    }
}